hack/build/common.sh

```diff
@@ -92,15 +115,46 @@ fi
 OUTPUT_DIR="${KERNEL_DIR}/target"
 mkdir -p "${OUTPUT_DIR}"
 
-KERNEL_CONFIG_FILE="${KERNEL_DIR}/configs/${TARGET_ARCH_STANDARD}/${KERNEL_FLAVOR}.config"
+mkdir -p "${KERNEL_OBJ}"
 
-if [ ! -f "${KERNEL_CONFIG_FILE}" ]; then
-	echo "ERROR: kernel config file not found for ${TARGET_ARCH_STANDARD}" >&2
-	exit 1
-fi
+# Copy out our custom kconfig - if we are building for a <flavor>-<variant>, merge the variant fragemnt with the flavor baseconfig first
+case "${KERNEL_FLAVOR}" in
+  *-*)
+    # Looks like we are dealing with <flavor>-<variant>.config, versus <flavor>.config
+    FLAVOR=$(echo "${KERNEL_FLAVOR}" | cut -d'-' -f1)
+    VARIANT=$(echo "${KERNEL_FLAVOR}" | cut -d'-' -f2)
+
+	if [ ! -f "${KERNEL_DIR}/configs/${TARGET_ARCH_STANDARD}/${FLAVOR}.config" ]; then
+		echo "ERROR: kernel flavor base config file not found for ${TARGET_ARCH_STANDARD}" >&2
+		exit 1
+	fi
+
+	if [ ! -f "${KERNEL_DIR}/configs/${TARGET_ARCH_STANDARD}/${FLAVOR}-${VARIANT}.fragment" ]; then
+		echo "ERROR: kernel flavor variant fragment config file not found for ${TARGET_ARCH_STANDARD}" >&2
+		exit 1
+	fi
+
+	if [ ! -f "${KERNEL_SRC}/scripts/kconfig/merge_config.sh" ]; then
+		echo "ERROR: cannot find upstream linux config merge script, is '$KERNEL_SRC' set?" >&2
+		exit 1
+	fi
+
+	# This will just drop a merged `.config` into ${KERNEL_OBJ}, which is what we want
+	sh "${KERNEL_SRC}"/scripts/kconfig/merge_config.sh -O "${KERNEL_OBJ}" "${KERNEL_DIR}/configs/${TARGET_ARCH_STANDARD}/${FLAVOR}.config" "${KERNEL_DIR}/configs/${TARGET_ARCH_STANDARD}/${FLAVOR}-${VARIANT}.fragment"
```

Comment:
Why not something like:

```sh
cp "${FLAVOR}.config" "${FLAVOR}-${VARIANT}.config" "${KERNEL_SRC}/arch/${TARGET_ARCH_STANDARD}/configs/"
make ARCH="${TARGET_ARCH_STANDARD}" defconfig "${FLAVOR}.config" "${FLAVOR}-${VARIANT}.config"
```

This is the officially-blessed way of merging kernel configuration fragments.

Reply:
Yep! I missed that I don't need to call `merge_configs.sh` directly, can just drop them in the right spots and the kernel `makefile` will handle it, ty!